crates/solvers/src/infra/config/baseline/file.rs

```diff
@@ -1,21 +1,17 @@
 use {
-    crate::{domain::eth, util::serialize},
+    crate::{domain::eth, infra::contracts, util::serialize},
+    ethereum_types::H160,
     serde::Deserialize,
     serde_with::serde_as,
     std::path::Path,
     tokio::fs,
 };
 
-#[serde_as]
 #[derive(Deserialize)]
 #[serde(rename_all = "kebab-case", deny_unknown_fields)]
 struct Config {
-    /// The chain ID the solver is configured for.
-    #[serde_as(as = "serialize::ChainId")]
-    pub chain_id: eth::ChainId,
-
-    /// The address of the WETH contract.
-    pub weth: Option<eth::H160>,
+    #[serde(flatten)]
+    pub contracts: ContractsConfig,
```

Comment:
Now we're mixing deny_unknown_fields and flatten which you mentioned before doesn't work. deny_unknown_fields has already been pretty helpful for me, but this is a small config file so doesn't really matter. More generally, I think not using flatten is nice because it makes the structure of the config file more obvious from the code. With complicated serde instructions it tends to get harder to figure out how the serialized file is going to look. Again, this file is small so not super important, so up to you.

Reply:
> Now we're mixing deny_unknown_fields and flatten which you mentioned before doesn't work

It doesn't work well. `deny_unknown_fields` and `flatten` together have a whole bunch of prickly issues:
- If you flatten a struct that also has `deny_unknown_fields`, then this won't work (the inner struct will see a whole bunch of fields from the parent and deny them).
- If you `deny_unknown_fields` on untagged enums, then this also [doesn't work](https://github.com/serde-rs/serde/issues/1600)

However, in this particular case of `deny_unknown_fields`, `flatten` and `rename` on an enum with only single values, it does. I tested it out locally, but you can also see an example of this [in the playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=ced40469267152bbf560a20719930bbb).

Reply:
So, I changed it to use a manual `match (chain_id, weth)`. The reasoning is:
- As you mentioned complicated `serde` instructions make things harder to figure out, also the whole complex interaction between `deny_unknown_fields`, `flatten` and `rename_all` is not obvious (for example, if you don't use the `rename_all`, then serialization will not work as expected)
- You get nicer panic error messages